tests/baselines/reference/computedPropertiesInDestructuring1.errors.txt

```diff
@@ -0,0 +1,88 @@
+tests/cases/compiler/computedPropertiesInDestructuring1.ts(33,23): error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+tests/cases/compiler/computedPropertiesInDestructuring1.ts(34,5): error TS2365: Operator '+' cannot be applied to types 'number' and '{}'.
+tests/cases/compiler/computedPropertiesInDestructuring1.ts(34,26): error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+
+
+==== tests/cases/compiler/computedPropertiesInDestructuring1.ts (16 errors) ====
+    // destructuring in variable declarations
+    let foo = "bar";
+    let {[foo]: bar} = {bar: "bar"};
+                        ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    
+    let {["bar"]: bar2} = {bar: "bar"};
+    
+    let foo2 = () => "bar";
+    let {[foo2()]: bar3} = {bar: "bar"};
+                            ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    
+    let [{[foo]: bar4}] = [{bar: "bar"}];
+                            ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    let [{[foo2()]: bar5}] = [{bar: "bar"}];
+                               ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    
+    function f1({["bar"]: x}: { bar: number }) {}
+    function f2({[foo]: x}: { bar: number }) {}
+    function f3({[foo2()]: x}: { bar: number }) {}
+    function f4([{[foo]: x}]: [{ bar: number }]) {}
+    function f5([{[foo2()]: x}]: [{ bar: number }]) {}
+    
+    // report errors on type errors in computed properties used in destructuring
+    let [{[foo()]: bar6}] = [{bar: "bar"}];
+           ~~~~~
+!!! error TS2349: Cannot invoke an expression whose type lacks a call signature.
+                              ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    let [{[foo.toExponential()]: bar7}] = [{bar: "bar"}];
+               ~~~~~~~~~~~~~
+!!! error TS2339: Property 'toExponential' does not exist on type 'string'.
+                                            ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    
+    // destructuring assignment
+    ({[foo]: bar} = {bar: "bar"});
+                     ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.
+    
+    ({["bar"]: bar2} = {bar: "bar"});
+    
+    ({[foo2()]: bar3} = {bar: "bar"});
+                         ~~~
+!!! error TS2353: Object literal may only specify known properties, and 'bar' does not exist in type '{}'.

```

Comment:
I agree with @DanielRosenwasser here. If the contextual type has computed property names it seems strange to restrict the source to only declared properties (which is practically guaranteed to not include the computed property). We should only perform the excess property check when there are no unknowns on the target side.